conans/test/integration/revisions_test.py

```diff
@@ -1,1062 +1,1452 @@
+        different RREV"""
+        pref = self.c_v2.create(self.ref, conanfile=GenConanfile().with_setting("os"),
+                                args="-s os=Windows")
+        pref2 = self.c_v2.create(self.ref, conanfile=GenConanfile().with_setting("os").
+                                 with_build_msg("I'm rev 2"),
+                                 args="-s os=Linux")
+
+        layout = self.c_v2.cache.package_layout(pref.ref.copy_clear_rev())
+
+        # Assert pref (outdated) is in the cache
+        self.assertTrue(layout.package_exists(pref.copy_clear_rev()))
+
+        # Assert pref2 is also in the cache
+        self.assertTrue(layout.package_exists(pref2.copy_clear_rev()))
+
+        self.c_v2.run("remove {} --outdated -f".format(pref.ref))
+
+        # Assert pref (outdated) is not in the cache anymore
+        self.assertFalse(layout.package_exists(pref.copy_clear_rev()))
+
+        # Assert pref2 is in the cache
+        self.assertTrue(layout.package_exists(pref2.copy_clear_rev()))
+
+    @parameterized.expand([(True,), (False,)])
+    def test_remove_oudated_packages_remote(self, v1):
+        """In a server with revisions uploaded no package is oudated so nothing is done, unless
+        a v1 upload mixed packages to a v1 or some hardcoded revision happen"""
+        self.c_v1.create(self.ref, conanfile=GenConanfile().
+                         with_setting("os").
+                         with_build_msg("I'm revision 1"),
+                         args="-s os=Windows")
+        self.c_v1.upload_all(self.ref)
+
+        # Different revision, different package_id (but everything uploaded to rev0)
+        self.c_v1.create(self.ref, conanfile=GenConanfile().
+                         with_setting("os").
+                         with_build_msg("I'm revision 2"),
+                         args="-s os=Linux")
+        self.c_v1.upload_all(self.ref)
+
+        # Verify in the server there is only one revision "0"
+        revs = self.server.server_store.get_recipe_revisions(self.ref)
+        self.assertEquals([r.revision for r in revs], [DEFAULT_REVISION_V1])
+
+        # Verify using v1 we can search for the outdated
+        data = self.c_v1.search(self.ref, remote="default", args="--outdated")
+        oss = [p["settings"]["os"] for p in data["results"][0]["items"][0]["packages"]]
+        self.assertEquals(set(["Windows"]), set(oss))
+        self.assertTrue(data["results"][0]["items"][0]["packages"][0]["outdated"])
+
+        # Verify we can remove it both using v1 or v2
+        client = self.c_v1 if v1 else self.c_v2
+        client.run("remove {} -r default --outdated -f".format(self.ref))
+
+        # The Windows package is not there anymore
+        data = self.c_v1.search(self.ref, remote="default", args="--outdated")
+        oss = [p["settings"]["os"] for p in data["results"][0]["items"][0]["packages"]]
+        self.assertEquals([], oss)
+
+        # But the Linux package is there, not outdated
+        data = self.
```

Comment:
I know that programatically it makes sense but the result could be a bit weird for users :weary: